sympy/assumptions/sathandlers.py

```diff
@@ -0,0 +1,376 @@
+from __future__ import print_function, division
+
+from collections import MutableMapping, defaultdict
+
+from sympy.core import (Add, Mul, Pow, Integer, Number, NumberSymbol,)
+from sympy.core.numbers import ImaginaryUnit
+from sympy.core.sympify import _sympify
+from sympy.core.rules import Transform
+from sympy.core.logic import fuzzy_or, fuzzy_and
+from sympy.matrices.expressions import MatMul
+
+from sympy.functions.elementary.complexes import Abs
+
+from sympy.assumptions.ask import Q
+from sympy.assumptions.assume import Predicate, AppliedPredicate
+from sympy.logic.boolalg import (Equivalent, Implies, And, Or,
+    BooleanFunction, Not)
+
+# APIs here may be subject to change
+
+# XXX: Better name?
+class UnevaluatedOnFree(BooleanFunction):
+    """
+    Represents a Boolean function that remains unevaluated on free predicates
+
+    This is intended to be a superclass of other classes, which define the
+    behavior on singly applied predicates.
+
+    A free predicate is a predicate that is not applied, or a combination
+    thereof. For example, Q.zero or Or(Q.positive, Q.negative).
+
+    A singly applied predicate is a free predicate applied everywhere to a
+    single expression. For instance, Q.zero(x) and Or(Q.positive(x*y),
+    Q.negative(x*y)) are singly applied, but Or(Q.positive(x), Q.negative(y))
+    and Or(Q.positive, Q.negative(y)) are not.
```

Comment:
I don't think so. I'm pointing out that this predicate is not singly applied because Q.positive is not applied and Q.negative(y) is.